Development/nmos/ocsp_utils.cpp

```diff
@@ -16,58 +16,65 @@ namespace nmos
                     }
-                    X509_ptr server_x509(PEM_read_bio_X509_AUX(server_bio.get(), NULL, NULL, NULL), &X509_free);
-                    if (!server_x509)
+                    X509_ptr issuer_x509(PEM_read_bio_X509_AUX(issuer_bio.get(), NULL, NULL, NULL), &X509_free);
+                    if (!issuer_x509)
                     {
-                        throw ocsp_exception("failed to make_ocsp_request while loading server certificate: PEM_read_bio_X509_AUX failure: " + ssl::experimental::last_openssl_error());
+                        throw ocsp_exception("failed to make_ocsp_request while loading issuer certificate: PEM_read_bio_X509_AUX failure: " + ssl::experimental::last_openssl_error());
                     }
 
-                    auto const cert_id_md = EVP_sha1();
-
-                    // creates and returns a new OCSP_CERTID structure using message digest dgst for certificate subject with issuer
-                    auto id = OCSP_cert_to_id(cert_id_md, server_x509.get(), issuer_x509.get());
-                    if (!id)
-                    {
-                        throw ocsp_exception("failed to make_ocsp_request while creating new OCSP_CERTID: OCSP_cert_to_id failure: " + ssl::experimental::last_openssl_error());
-                    }
-                    if (!OCSP_request_add0_id(ocsp_req.get(), id))
+                    const auto& server_certificates = issuer_certificate_vs_server_certifiactes_item.second;
+                    // load server certificate then add to OCSP request
+                    for (const auto& server_certificate : server_certificates)
                     {
-                        throw ocsp_exception("failed to make_ocsp_request while adding certificate to OCSP request: OCSP_request_add0_id failure: " + ssl::experimental::last_openssl_error());
+                        // load server certificate
+                        BIO_ptr server_bio(BIO_new(BIO_s_mem()), &BIO_free);
+                        if ((size_t)BIO_write(server_bio.get(), server_certificate.data(), (int)server_certificate.size()) != server_certificate.size())
+                        {
+                            throw ocsp_exception("failed to make_ocsp_request while loading server certificate: BIO_write failure: " + ssl::experimental::last_openssl_error());
+                        }
+                        X509_ptr server_x509(PEM_read_bio_X509_AUX(server_bio.get(), NULL, NULL, NULL), &X509_free);
+                        if (!server_x509)
+                        {
+                            throw ocsp_exception("failed to make_ocsp_request while loading server certificate: PEM_read_bio_X509_AUX failure: " + ssl::experimental::last_openssl_error());
+                        }
+
+                        auto const cert_id_md = EVP_sha1();
+
+                        // creates and returns a new OCSP_CERTID structure using message digest dgst for certificate subject with issuer
```

Comment:
Copied sentence from man page didn't quite make sense to me...

```suggestion
                        // creates and returns a new OCSP_CERTID structure using SHA1 message digest for the server certificate with the issuer
```